.github/workflows/xe-experimental-vm-test.yml

```diff
@@ -0,0 +1,35 @@
+name: "[experimental] VM-based integration testing, ping @Xe if it fails"
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - 'main'
+  release:
+    types: [ created ]
+  schedule:
+    - cron: '00 6,18 * * *'
+
+jobs:
+  experimental-linux-vm-test:
+    runs-on: [ self-hosted, linux, vm_integration_test ]
+
+    if: "!contains(github.event.head_commit.message, '[ci skip]')"
+
+    steps:
```

Comment:
Don't you need to force Go 1.16 like Linux does?

Reply:
No, the github runner is configured to have nixpkgs Go already in its path with a NixOS module that looks like this:

```nix
{ lib, config, pkgs, ... }:

{
  services.github-runner = {
    enable = true;

    url = "https://github.com/tailscale/tailscale";
    name = "xe-tstest-integration-vms";
    replace = true;
    tokenFile = "/run/decrypted/ts-oss-ghaction-token";
    extraLabels = [ "vm_integration_test" ];
    extraPackages = with pkgs; [ go cdrkit qemu openssh gcc ];
  };

  users.users.ghrunner = {
    createHome = true;
    isSystemUser = true;
    extraGroups = [ "kvm" ];
  };

  systemd.services.github-runner = {
    serviceConfig = {
      DeviceAllow = lib.mkForce [ "/dev/kvm" ];
      DynamicUser = lib.mkForce false;
      PrivateDevices = lib.mkForce false;
      ProtectSystem = lib.mkForce null;
      User = "ghrunner";
      ExtraGroups = [ "kvm" ];
    };
  };
}
```

Reply:
That's a little indirect. Check in the nix configs too, then, and add a comment in the github action yml referencing the nix configs, so it's not so implicit?

Reply:
I added `runner.nix` (a much more commented version of that file) to tstest/integration/vms